ddtrace/utils/hook.py

```diff
@@ -0,0 +1,196 @@
+        notify_module_loaded(module)
+        return module
+
+
+class ImportHookFinder:
+    def __init__(self):
+        self.in_progress = {}
+
+    @synchronized(_post_import_hooks_lock)
+    def find_module(self, fullname, path=None):
+        # If the module being imported is not one we have registered
+        # post import hooks for, we can return immediately. We will
+        # take no further part in the importing of this module.
+
+        if fullname not in _post_import_hooks:
+            return None
+
+        # When we are interested in a specific module, we will call back
+        # into the import system a second time to defer to the import
+        # finder that is supposed to handle the importing of the module.
+        # We set an in progress flag for the target module so that on
+        # the second time through we don't trigger another call back
+        # into the import system and cause a infinite loop.
+
+        if fullname in self.in_progress:
+            return None
+
+        self.in_progress[fullname] = True
+
+        # Now call back into the import system again.
+
+        try:
+            if PY3:
+                # For Python 3 we need to use find_spec().loader
+                # from the importlib.util module. It doesn't actually
+                # import the target module and only finds the
+                # loader. If a loader is found, we need to return
+                # our own loader which will then in turn call the
+                # real loader to import the module and invoke the
+                # post import hooks.
+                try:
+                    import importlib.util
+                    loader = importlib.util.find_spec(fullname).loader
+                except (ImportError, AttributeError):
+                    loader = importlib.find_loader(fullname, path)
+                if loader:
+                    return _ImportHookChainedLoader(loader)
+            else:
+                # For Python 2 we don't have much choice but to
+                # call back in to __import__(). This will
+                # actually cause the module to be imported. If no
+                # module could be found then ImportError will be
+                # raised. Otherwise we return a loader which
+                # returns the already loaded module and invokes
+                # the post import hooks.
+                __import__(fullname)
+                return _ImportHookLoader()
+
+        finally:
+            del self.in_progress[fullname]
+
+
+@synchronized(_post_import_hooks_lock)
+def deregister_post_import_hook(modulename, hook):
+    """
+    Deregisters post import hooks for a module given the module name and a hook
+    that was previously installed.
+
+    :param modulename: Name of the module the hook is installed on.
+    :type: str
+    :param hook: The hook to remove (the function itself)
+    :type hook: Callable
+    :return: whether a hook was removed or not
```

Comment:
```suggestion
    :return: whether a hook was removed or not
    :rtype: bool
```

Reply:
aw, I can't accept my own suggestion :(

This doesn't matter, I don't really want to have to do a commit and push and whatever...